swiftplugin/swiftplugin/static/js/src/swiftplugin.js

```diff
@@ -9,32 +9,26 @@ function SwiftPluginXBlock(runtime, element) {
     var solutionCodeMirror = null;
 
     const run_btn = document.getElementById('run-btn');
-
-    function getCodeAndMode() {
+    run_btn.onclick = function (eventObject) {
         var user_code = myCodeMirror.getValue()
         var code_mode = myCodeMirror.getMode()
         var mode = code_mode.helperType ? code_mode.helperType : code_mode.name;
-        return {user_code, mode};
-    }
-
-    run_btn.onclick = function (eventObject) {
-        const {user_code, mode} = getCodeAndMode();
         $.ajax({
             type: "POST",
             url: handlerUrl,
-            data: JSON.stringify({type: 'run', code: user_code, language: mode}),
+            data: JSON.stringify({ type: 'run', code: user_code, language: mode }),
             success: updateResponse,
             error: handleError
         });
     }
 
     const submit_btn = document.getElementById('submit-btn');
     submit_btn.onclick = function (eventObject) {
-        const {user_code, mode} = getCodeAndMode();
+        var user_code = myCodeMirror.getValue();
```

Comment:
why are these lines changed?

Reply:
Please ignore them. I was experimenting with something, and later forgot to remove these lines.

Reply:
Okay, lets remove this line. PR looks good. Once these lines are removed I can give it a final test